x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderDecisionTests.java

```diff
@@ -381,16 +381,40 @@ private static void verifyScale(ClusterState state, long expectedDifference, Str
         );
         TestAutoscalingDeciderContext context = createContext(state, Set.of(DiscoveryNodeRole.DATA_HOT_NODE_ROLE));
         AutoscalingDeciderResult result = decider.scale(Settings.EMPTY, context);
+        ReactiveStorageDeciderService.ReactiveReason resultReason = (ReactiveStorageDeciderService.ReactiveReason) result.reason();
+
         if (context.currentCapacity != null) {
             assertThat(
                 result.requiredCapacity().total().storage().getBytes() - context.currentCapacity.total().storage().getBytes(),
                 equalTo(expectedDifference)
             );
-            assertThat(result.reason().summary(), equalTo(reason));
+            assertThat(resultReason.summary(), equalTo(reason));
         } else {
             assertThat(result.requiredCapacity(), is(nullValue()));
-            assertThat(result.reason().summary(), equalTo("current capacity not available"));
+            assertThat(resultReason.summary(), equalTo("current capacity not available"));
         }
+        assertThat(
```

Comment:
Can we add a more explicit test case? i.e. when a shard moves from hot to warm and that triggers the autoscaling decision.

Reply:
I guess it's a bit tricky, because it's a generic `verify` method which gets invoked from multiple tests. I will see if I can create a separate test.

Reply:
I think it's covered in `testStoragePreventsAllocation`. Can we verify there that we see the "unassigned" shards there?

Reply:
Added via 

```java
assertThat(resultReason.unassignedShardIds(), equalTo(decider.allocationState(context).storagePreventsAllocation().shardIds()))
```